### DISCUSSION TITLE
Acoplamiento de observer

### DISCUSSION THREAD

**Discussion by Franpanozzo** [dds-jv/foro#296] [Consultas de objetos] [answered]
Buenas, tengo una duda rápida sobre una respuesta que vi en un parcial por ahí dando vueltas. Teniendo el siguiente diagrama de clases:

![DudaPGMPARCIAL2](https://user-images.githubusercontent.com/62455934/124703874-7d7bf300-dec9-11eb-9aec-1ef48876ed6d.PNG)

El que resolvió el parcial pone que: **_en esta solución el hormiguero con la hormiga no esta tan acoplado porque no se conocen mutuamente_**, se lo marcaron como bien. Ahora los observers para poder mandarle el mensaje al hormiguero por ejemplo de que mandeHormigas tienen que de algun lado conocerlo. Entonces la hormiga no tendría tambien que conocer a que hormiguero pertenece? Y así poder pasárselo a los observers. Saludos!

**Comment by gastonprieto** [comment, marked as answer]
Buenas @Franpanozzo 

Es importante antes de poder responderte que nos pases el enunciado + solución de lo que leíste, porque depende de lo que se haya escrito en la solución y tal vez de otras afirmaciones, que eso en ese contexto sea correcto. En particular lo que marcamos como bien/regular/mal depende del contexto, se evaluan en conjunto en base a los requerimientos y al planteamiento global de la solución. Así que es un poco decirte porque en ese caso era correcto.

Por otro lado igual te comento que no necesariamente todo lo que usan los objetos lo van a recibir por parametro, sino que también pueden usar objetos que ya esten en su estado interno, eso depende si es una dependencia que va cambiando según la invocación o una dependencia que esta mas atada al objeto. Por ejemplo tal vez el observer MandarHormigas cuando lo instanciaron le pasaron el hormiguero (aparentemente es esto según la flecha de asociación que tiene) entonces la hormiga no tenia que conocerlo para pasarselo por parametro.

**Reply by Franpanozzo** [reply]
Perfecto, lo había pensado en un momento pero no me cerraba, que a las clases previamente se las instancie con el hormiguero, ahora pensándolo mejor con lo que me dijiste me cerro, gracias Gastón!
